src/contracts/atlas/GasAccounting.sol

```diff
@@ -85,29 +93,31 @@ abstract contract GasAccounting is SafetyLocks {
         if (_lock.activeEnvironment != environment) {
             revert InvalidExecutionEnvironment(_lock.activeEnvironment);
         }
```

Comment:
```suggestion
```
Do we still need to pass the Execution Environment address in `reconcile()`? The function can only be called during SolverOperation, and the only thing that happens in that phase is Atlas calls directly to the solver, so the solver contract would need to call to a malicious contract to give it the opportunity to call `reconcile()`. Also, supplying the correct EE address as a param is trivial as it can be read off Atlas via `lock()`, so its not really a protective measure.

I think we can remove this check and the `environment` completely to save gas. The phase check acts as a "not unlocked" check as well.

Reply:
good catch! I agree. 

The one argument for keeping it would be to help solvers not get rugged - this check effectively lets them verify that they're sending the bids to the correct ExecutionEnvironment. But that's not a major issue due to the fact that Atlas is now the msg.sender anyway. Let's make this change.

Reply:
Fixed here https://github.com/FastLane-Labs/atlas/pull/271/commits/f2d5b2e8d7387fffdff4378e02108bd5ef596e10